bindings/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+option(FRAMEWORK_USES_PYTHON "Do you want to create the Python bindings" FALSE)
+
+if(FRAMEWORK_USES_PYTHON)
+
+    # Find the necessary Python libraries
+    find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+    message(STATUS "Using Python: ${Python3_EXECUTABLE}")
+
+    # Install the resulting Python package for the active interpreter
+    if(NOT PYTHON_INSTDIR)
+        set(PYTHON_INSTDIR ${Python3_SITELIB}/bipedal_locomotion)
+    endif()
```

Comment:
Note: this by default will install files **outside**  the `CMAKE_INSTALL_PREFIX` if `FRAMEWORK_USES_PYTHON` is set to `ON` , at the superbuild level we will deal with this by pasing `PYTHON_INSTDIR`, but I want to highlight this as it may be idiomatic for Python folks, but it may surprise C++ people @GiulioRomualdi @S-Dafarra @prashanthr05

Reply:
Thanks for making this clear to everyone. This default would result in a importable Python package without the need of operating on the system environment, and I think this is a sane default. Then, if the user wants more control about where the Python package is installed, paying the price of manually configuring their `PYTHON_PATH`, the CMake project allows to alter this default by passing `-DPYTHON_INSTDIR=/my/custom/destination/for/bipedal_locomotion`.

Reply:
> This default would result in a importable Python package without the need of operating on the system environment, and I think this is a sane default. 

As someone that debugged a lot of issues caused by `export(PACKAGE)` that was installing stuff outside the `CMAKE_INSTALL_PREFIX` by default (but inside a user-readable directory), I fear that this will create more problems then it solves, but it is a tradeoff choice that I would leave to the maintainers.

Reply:
From my point of view a decently documented project should prevent these headaches of external packagers, but if I have to chose between what I personally think a simpler use-case for regular users and a particular need of the few packagers (that I bless their existence and I'd like to hug them all for what they do), I'd always tend to prefer the former. Headaches come from undocumented and rather odd and hidden operations, I think that hybrid C++/Python projects by default need extra packaging care. 

This being said, I'll wait the opinion of the maintainers of the project.

Reply:
Sorry, I probably was not clear: I do not think that this behavior is problematic for packagers, I think it is problematic and error-prone for regular users that install the library manually via C++ by following the instructions on the README. 
With the default `CMAKE_INSTALL_PREFIX`, this results in some files installed in `/usr/local` (like the vast majority of user expect) but some files will get installed **by using the default settings** in `/usr`, getting mixed with apt-installed files and making it more complex to eventually cleanup the library if a system is dirty. Similarly, if the `CMAKE_INSTALL_PREFIX` is modified to point to a user directory (as usually done by CMake/C++ users) all the C++ files will be installed in the directory specified by `CMAKE_INSTALL_PREFIX`, while just the python stuff will still go in `/usr`. Similar problem will arise (with the default settings) if you try to build the library with colcon, a tool extremely used in robotics to build Python/C++ mixed projects. 

As you were mentioning before, it does not make a lot of sense to simplify life for a small minority of the users if this breaks [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) for the majority of the target users of this library. And in this case I think that "Python users that use virtual envs" are the minority, while users that are used to how robotics libraries and their python bindings are installed (such as ROS/YARP/iDynTree/Pinocchio) are the majority, but clearly this is just my personal non-scientific evaluation.

Reply:
Sorry if I misunderstood, now your point is more clear.

> With the default `CMAKE_INSTALL_PREFIX`, this results in some files installed in `/usr/local` (like the vast majority of user expect) but some files will get installed **by using the default settings** in `/usr`.

This is correct, by default the Python tree located in `/usr/lib/python3/dist-packages` is detected by CMake's `FindPython3`.

> As you were mentioning before, it does not make a lot of sense to simplify life for a small minority of the users if this breaks principle of least astonishment for the majority of the target users of this library. And in this case I think that "Python users that use virtual envs" are the minority, while users that are used to how robotics libraries and their python bindings are installed (such as ROS/YARP/iDynTree/Pinocchio) are the majority, but clearly this is just my personal non-scientific evaluation.

From my point of view, flipping the behaviour is not a problem. I believe that the current status of how the libraries typically handle their Python files is an inheritance due to the lack of support from tools like CMake, that greatly improved in the near past. Then, I also believe that Python started being more used also in robotics recently, and yet we did not converge to an uniform approach. On the projects I maintain, I'll keep this behaviour as default, I leave the last word to the maintainer of this project @S-Dafarra @GiulioRomualdi. Introducing a new CMake option that allows installing outside the `CMAKE_INSTALL_PREFIX`, disabled by default, would be a trivial change.

---

> getting mixed with apt-installed files and making it more complex to eventually cleanup the library if a system is dirty

I never though about what I'm going to say, but a simple `sudo pip uninstall <python_package>` would suffice to remove what got installed in `/usr`. A dirty `/usr/local` with many different packages installed from source is anyway very difficult to clean up from a single package, if you need to do that most likely means that something really wrong is going on.

Reply:
Personal opinion: I would first popularize the new method (i.e. installing Python files outside `CMAKE_INSTALL_PREFIX`) at least in our internal DIC users, and then (even indeed it starts get used) I would switch the default to use the new method, not the other way around.

Reply:
Done in ebaf487